go/http/httpbase.go

```diff
@@ -27,20 +28,39 @@ import (
 	"github.com/github/orchestrator/go/inst"
 	"github.com/github/orchestrator/go/os"
 	"github.com/github/orchestrator/go/process"
-	"github.com/github/orchestrator/go/raft"
+	orcraft "github.com/github/orchestrator/go/raft"
+	"github.com/openark/golib/log"
 )
 
+const regexInitRequired = "#needscompile"
+
+var regexURI = regexp.MustCompile(regexInitRequired)
+
 func getProxyAuthUser(req *http.Request) string {
 	for _, user := range req.Header[config.Config.AuthUserHeader] {
 		return user
 	}
 	return ""
 }
 
+// isForcedReadOnly checks config to see if the WebInterfaceReadOnly flag is set
+// This is independent of any authentication mechanisms
+func isForcedReadOnly(uri string) bool {
+	if fmt.Sprintf("%v", regexURI) == regexInitRequired {
+		log.Debug("Compiling regexp for isForcedReadOnly")
+		regexURI = regexp.MustCompile(fmt.Sprintf("^%s/web/.*", config.Config.URLPrefix))
+	}
```

Comment:
Can you please clarify the use of the regex here? I confess I do not understand what this code is doing.
Also, please note `regexURI` is a global variable, which is being compared to, and manipulated, in `isForcedReadOnly(...)`, which gets called by `isAuthorizedForAction(...)`, which gets called by various API/Web endpoints, potentially concurrently. Thus, regardless of the purpose of the logic above, it is incorrect seeing that multiple concurrent calls to API/Web will corrupt the value of `regexURI`.

Reply:
So, the regex is intended to explicitly match the configured URLPrefix, but that is not available at compile time. Instead of repeatedly compiling a regex, I was looking to have it set correctly by the first call that triggers it. Are you happy with a broader match?
```
var regexURI = regexp.MustCompile("^.*/?web/.*")
```